superset-frontend/spec/helpers/testing-library.tsx

```diff
@@ -81,8 +95,8 @@ export function createWrapper(options?: Options) {
       result = <BrowserRouter>{result}</BrowserRouter>;
     }
 
-    if (useQuery) {
-      result = <QueryProvider>{result}</QueryProvider>;
+    if (useQuery && !useRedux) {
```

Comment:
Previously, the developer had the option to use query params without Redux. Now it's a little bit weird because you are setting `useQuery` and the store is being configured underneath. `useQuery` is more related to the URL query params than Redux. Could you separate the concepts again?

Reply:
oh it's not `useQueryParams`. (it's still [there](https://github.com/apache/superset/pull/23760/files/9dda4c6f14b8c8fe02b52bb445fe30fa376a833c#diff-003094fe62e8b418be60870591fb179b47fae671e2376a784c62ac4148b6c0eaL76)) the name looks confuse. I'll replace the name with `useApi` then.